local_vars/local_vars.cc

```diff
@@ -418,42 +420,44 @@ class LocalNameInserter {
             } else {
                 boxedKwArgs = ast::MK::Nil(original.loc);
             }
-            original.addPosArg(std::move(boxedKwArgs));
+            newArgs.push_back(std::move(boxedKwArgs));
+            newNumPosArgs = newArgs.size();
 
             if (originalBlock != nullptr) {
                 // <call-with-splat> and "do"
                 newFun = core::Names::callWithSplat();
                 // Re-add block argument
                 newFlags.hasBlock = true;
-                original.setBlock(std::move(originalBlock));
+                newArgs.push_back(std::move(originalBlock));
             } else if (shouldForwardBlockArg) {
                 // <call-with-splat-and-block>(..., &blk)
                 newFun = core::Names::callWithSplatAndBlock();
-                original.addPosArg(std::move(blockArg));
+                newArgs.push_back(std::move(blockArg));
+                newNumPosArgs++;
             } else {
                 // <call-with-splat>(...)
                 newFun = core::Names::callWithSplat();
             }
         } else if (originalBlock == nullptr && shouldForwardBlockArg) {
-            // No positional splat and no "do", so we need to forward &<blkvar> with <call-with-block>.
-            original.reserveArguments(3 + posArgsEntries.size(), kwArgKeyEntries.size(),
```

Comment:
You can delete the definition of `reserveArguments` from `Trees.h`, these are the only 2 uses of it.